pokemongo_bot/cell_workers/catch_lured_pokemon.py

```diff
@@ -1,36 +1,47 @@
+import time
+
 from pokemongo_bot import logger
-from pokemongo_bot.cell_workers.utils import fort_details
+from pokemongo_bot.constants import Constants
+from pokemongo_bot.cell_workers.utils import fort_details, distance
 from pokemongo_bot.cell_workers.pokemon_catch_worker import PokemonCatchWorker
 from pokemongo_bot.cell_workers.base_task import BaseTask
 
 class CatchLuredPokemon(BaseTask):
-    def work(self):
+    def work(self, *args, **kwargs):
         lured_pokemon = self.get_lured_pokemon()
         if lured_pokemon:
             self.catch_pokemon(lured_pokemon)
 
     def get_lured_pokemon(self):
         forts = self.bot.get_forts(order_by_distance=True)
 
-        if len(forts) == 0:
-            return False
-
-        fort = forts[0]
-        details = fort_details(self.bot, fort_id=fort['id'],
-                              latitude=fort['latitude'],
-                              longitude=fort['longitude'])
-        fort_name = details.get('name', 'Unknown').encode('utf8', 'replace')
-
-        encounter_id = fort.get('lure_info', {}).get('encounter_id', None)
-
-        if encounter_id:
-            logger.log('Lured pokemon at fort {}'.format(fort_name))
-            return {
-                'encounter_id': encounter_id,
-                'fort_id': fort['id'],
-                'latitude': fort['latitude'],
-                'longitude': fort['longitude']
-            }
+        forts_in_range = []
+
+        for fort in forts:
+            distance_to_fort = distance(
+                self.bot.position[0],
+                self.bot.position[1],
+                fort['latitude'],
+                fort['longitude']
+            )
+
+            if distance_to_fort <= Constants.MAX_DISTANCE_FORT_IS_REACHABLE:
+                forts_in_range.append(fort)
+
+        forts_with_lured_pokemons = [fort for fort in forts_in_range if fort.get('lure_info', {}).get('encounter_id', None)]
+
+        for fort in forts_with_lured_pokemons:
+            encounter_id = fort.get('lure_info', {}).get('encounter_id', None)
+
+            if encounter_id:
+                self.catch_pokemon({
+                    'encounter_id': encounter_id,
+                    'fort_id': fort['id'],
+                    'latitude': fort['latitude'],
+                    'longitude': fort['longitude']
+                })
+
+            time.sleep(1)
```

Comment:
Yes, when we got the bot working again, we'll do it right and remove the [XGH](https://gist.github.com/banaslee/4147370) code. 😄